packages/generate-persisted-query-manifest/src/index.ts

```diff
@@ -19,41 +19,64 @@ import { first, sortBy } from "lodash";
 import { createHash } from "node:crypto";
 import { readFileSync } from "node:fs";
 
-export interface GeneratePersistedQueryManifestOptions {
-  files?: string[];
+type RequireKeys<T, K extends keyof T> = Required<Pick<T, K>> & Omit<T, K>;
 
-  globInclude?: string[];
-  globExclude?: string[];
+export interface PersistedQueryManifestConfig {
+  /**
+   * Paths to your GraphQL documents: queries, mutations, subscriptions, and fragments.
+   */
+  documents?: string | string[];
+
+  /**
+   * Paths that should be ignored when searching for your GraphQL documents.
+   */
+  documentIgnorePatterns?: string | string[];
+
+  /**
+   * Path where the manifest file will be written.
+   */
+  output?: string;
 }
 
+type DefaultPersistedQueryManifestConfig = RequireKeys<
+  PersistedQueryManifestConfig,
+  "documents" | "documentIgnorePatterns" | "output"
+>;
+
 export interface PersistedQueryManifestOperation {
   id: string;
   name: string;
   type: "query" | "mutation" | "subscription";
   body: string;
 }
+
 export interface PersistedQueryManifest {
   format: "apollo-persisted-query-manifest";
   version: 1;
   operations: PersistedQueryManifestOperation[];
 }
 
-export async function generatePersistedQueryManifest(
-  options: GeneratePersistedQueryManifestOptions = {},
-): Promise<PersistedQueryManifest> {
-  const files = new Set<string>(options.files ?? []);
-  const globInclude = options.globInclude ?? [
-    "src/**/*.{graphql,tsx,jsx,ts,js}",
-  ];
-  const globExclude = options.globExclude ?? [
+export const defaults: DefaultPersistedQueryManifestConfig = {
+  documents: "src/**/*.{graphql,js,jsx,ts,tsx}",
```

Comment:
Let's get .gql in the defaults too

Reply:
good call!

Reply:
Added in [1e8a07d](https://github.com/apollographql/apollo-utils/pull/295/commits/1e8a07d657f2a6c3dde7eec68d941c771218f1f5)